src/main/java/SequenceValidator.java

```diff
@@ -5,167 +5,376 @@
+            if (courseOccurences.size() > 1) {
+                sequenceIsValid = false;
+
+                JSONArray positions = new JSONArray();
+                for (Point courseOccurencePosition : courseOccurences) {
+                    int uniqueSemesterId = (int) courseOccurencePosition.getX(); // no precision lost since
+                    int courseIndex = (int) courseOccurencePosition.getY(); // double has 52 bits and int has 32
+                    positions.put(positionObject(uniqueSemesterId, courseIndex));
+                }
 
-        if(issues.size() > 0){
-            responseMessage.put("valid","false");
-            for(SequenceIssue issue:issues){
-                issueArray.put(issue.toJSONObject());
+                warnings
+                .put(new JSONObject()
+                    .put("type", WARNINGS.REPEATED.toString())
+                    .put("data", new JSONObject()
+                        .put("courseCode", courseCode)
+                        .put("positions", positions)
+                    )
+                );
             }
-            responseMessage.put("issues", issueArray);
         }
 
-        return responseMessage;
+        // CREDIT COUNT
+
+        if (creditCount < minTotalCredits) {
+            sequenceIsValid = false;
+
+            issues
+            .put(new JSONObject()
+                .put("type", ISSUES.CREDITCOUNT.toString())
+                .put("data", new JSONObject()
+                    .put("required", minTotalCredits.toString())
+                    .put("actual", creditCount.toString())
+                )
+            );
+        }
+
+        // SEND ISSUE LIST IF SEQUENCE NOT VALID
+
+        if (!sequenceIsValid) {
+            sequenceValidationResults
+            .put("isValid", "false")
+            .put("issues", issues)
+            .put("warnings", warnings);
+        } else {
+            sequenceValidationResults.put("isValid", "true");
+        }
+
+        return sequenceValidationResults;
+    }
+
+    // helpers to create JSON object representing position of course in course sequence
+    private JSONObject positionObject(Integer yearIdx, Integer seasonIdx, Integer courseIdx) throws JSONException {
+        return new JSONObject()
+            .put("yearIndex", yearIdx.toString())
+            .put("season", SEASONS.values()[seasonIdx].toString())
+            .put("courseIndex", courseIdx.toString());
+    }
+    private JSONObject positionObject(Integer uniqueSemesterId, Integer courseIdx) throws JSONException {
+        return new JSONObject()
+            .put("yearIndex", new Integer(uniqueSemesterId / SEASONS.values().length).toString())
+            .put("season", SEASONS.values()[uniqueSemesterId % SEASONS.values().length].toString())
+            .put("courseIndex", courseIdx.toString());
+    }
+    // helper to extract position from JSON position object
+    private Point getPosFromPosistionObject(JSONObject positionObject) throws JSONException {
```

Comment:
Don't know if you left this here intentionally or not. If yes then it's fine by me.

Reply:
If this method is unused, I'll remove it.  You can expect all the above changes to be implemented by tonight.  I will merge the PR myself after the last aforementioned change is made.

Reply:
Kk